src/components/ModalEmailHOC.vue

```diff
@@ -0,0 +1,14 @@
+<script lang="ts">
+export default {
+  inheritAttrs: false
+};
+</script>
+
+<script setup lang="ts">
+const { isSubscribed } = useEmailSubscription();
+</script>
+
+<template>
+  <ModalEmailSubscription v-if="!isSubscribed" v-bind="$attrs" />
```

Comment:
Why do you need to use `$attrs` here, can you use props? 

Should also have a `div` as root in template